erizo/src/test/NiceConnectionTest.cpp

```diff
@@ -0,0 +1,426 @@
+    WillOnce(DoAll(SetArgPointee<2>(ufrag), SetArgPointee<3>(pass), Return(true)));
+  EXPECT_CALL(*libnice, NiceAgentAttachRecv(_, _, _, _, _, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentGatherCandidates(_, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentSetRemoteCredentials(_, _, _, _)).Times(0);
+  EXPECT_CALL(*libnice, NiceAgentSetPortRange(_, _, _, 10, 60)).Times(1);
+  EXPECT_CALL(*libnice, NiceAgentSetRelayInfo(_, _, _, _, _, _, _)).Times(0);
+
+  erizo::NiceConnection nice(libnice_pointer,
+      erizo::VIDEO_TYPE,
+      "video",
+      "test_connection",
+      nice_listener,
+      1,
+      *ice_config);
+  nice.start();
+}
+
+TEST_F(NiceConnectionStartTest, start_Configures_Libnice_With_Turn) {
+  char *ufrag = strdup("ufrag");
+  char *pass = strdup("pass");
+
+  ice_config->turnServer = "atunrserver.com";
+  ice_config->turnPort = 3478;
+  ice_config->turnUsername = "test";
+  ice_config->turnPass = "test";
+
+  EXPECT_CALL(*libnice, NiceAgentNew(_)).Times(1);
+  EXPECT_CALL(*libnice, NiceAgentAddStream(_, _)).Times(1);
+  EXPECT_CALL(*libnice, NiceAgentGetLocalCredentials(_, _, _, _)).Times(1).
+    WillOnce(DoAll(SetArgPointee<2>(ufrag), SetArgPointee<3>(pass), Return(true)));
+  EXPECT_CALL(*libnice, NiceAgentAttachRecv(_, _, _, _, _, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentGatherCandidates(_, _)).Times(1).WillOnce(Return(true));
+  EXPECT_CALL(*libnice, NiceAgentSetRemoteCredentials(_, _, _, _)).Times(0);
+  EXPECT_CALL(*libnice, NiceAgentSetPortRange(_, _, _, _, _)).Times(0);
+  EXPECT_CALL(*libnice, NiceAgentSetRelayInfo(_, _, _, ice_config->turnServer.c_str(),
+        ice_config->turnPort, ice_config->turnUsername.c_str(),
+        ice_config->turnPass.c_str())).Times(1).
+        WillOnce(Return(true));
+
+  erizo::NiceConnection nice(libnice_pointer,
+      erizo::VIDEO_TYPE,
+      "video",
+      "test_connection",
+      nice_listener,
+      1,
+      *ice_config);
+  nice.start();
+}
+
+TEST_F(NiceConnectionTest, setRemoteCandidates_Success_WhenCalled) {
+  erizo::CandidateInfo test_candidate;
+  test_candidate.isBundle = true;
+  test_candidate.priority = 0;
+  test_candidate.componentId = 1;
+  test_candidate.foundation = "10";
+  test_candidate.hostAddress = "192.168.1.1";
+  test_candidate.rAddress = "";
+  test_candidate.hostPort = 5000;
+  test_candidate.rPort = 0;
+  test_candidate.netProtocol = "udp";
+  test_candidate.hostType = erizo::HOST;
+  test_candidate.username = "hola";
+  test_candidate.password = "hola";
+  test_candidate.mediaType = erizo::VIDEO_TYPE;
+
+  std::vector<erizo::CandidateInfo> candidate_list;
+  candidate_list.push_back(test_candidate);
+
+  EXPECT_CALL(*libnice, NiceAgentSetRemoteCandidates(_, _, _, _)).Times(1);
+  nice_connection->setRemoteCandidates(candidate_list, true);
+}
+
+TEST_F(NiceConnectionTest, queuePacket_QueuedPackets_Can_Be_getPacket_When_Ready) {
+  char* test_packet = strdup("tester");
```

Comment:
maybe it makes sense to move test_packet creation and deletion to the NiceConnectionTest since you're using it in more tests.